### PR TITLE
Add 3DS configuration

### PR DESCRIPTION
**Pull request by zoeyjodon**
Changelog:
- Adds unix definitions for the 3DS
- Disables AF_INET6 behavior for 3DS builds
- Guards timeout behavior for 3DS builds
- Removes trailing whitespace

### CODE REVIEW

File: unix.c [hunk 1]
```diff
@@ -201,7 +232,7 @@ enet_address_equal (ENetAddress * address1, ENetAddress * address2)
         return sin1 -> sin_port == sin2 -> sin_port &&
             sin1 -> sin_addr.s_addr == sin2 -> sin_addr.s_addr;
     }
-#ifdef AF_INET6
+#if defined(AF_INET6) && !(defined(__3DS__))
```
Comment (cgutman) [inline unix.c]: What's missing in the 3DS SDK? Do they define `AF_INET6` but not `struct sockaddr_in6` or something?

Comment (zoeyjodon) [reply unix.c]: Yup! I can only really find references to `sockaddr_in6` in devkitArm's "experimental" code, but it's not actually defined in the toolkit.

Comment (cgutman) [reply unix.c]: OK, you can revert this part then since you already `#undef AF_INET6` above.

Comment (zoeyjodon) [reply unix.c]: Done! Sorry, I must've forgotten to remove those when I added the `undef` bits.

**Comment by cgutman** [comment]
Merged, thanks!

**Pull request merged**